### PR TITLE
Add mfast recipe

### PR DESCRIPTION
**Pull request by tarc**
Specify library name and version:  **mfast/1.2.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mfast/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class mFASTConan(ConanFile):
+    name = "mfast"
+    Name = "mFAST"
```
Comment (uilianries) [inline recipes/mfast/all/conanfile.py:7]:
```suggestion
```

Don't need to re-use it.

File: recipes/mfast/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.Name + "-" + self.version, self._source_subfolder)
```
Comment (uilianries) [inline recipes/mfast/all/conanfile.py:41]:
```suggestion
        os.rename("mFAST-" + self.version, self._source_subfolder)
```

File: recipes/mfast/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("mFAST-" + self.version, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```
Comment (uilianries) [inline recipes/mfast/all/conanfile.py:45]:
```suggestion
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC
```

File: recipes/mfast/all/patches/00005-add-conan-libs-to-targets.patch.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+diff --git a/src/fast_type_gen/CMakeLists.txt b/src/fast_type_gen/CMakeLists.txt
+index 4e604cd..3762ea3 100644
+--- a/src/fast_type_gen/CMakeLists.txt
++++ b/src/fast_type_gen/CMakeLists.txt
+@@ -11,8 +11,7 @@ add_executable (fast_type_gen
+ target_link_libraries (fast_type_gen
+                        mfast_xml_parser_static
+                        mfast_static
+-                       ${Boost_SYSTEM_LIBRARY}
```
Comment (uilianries) [inline recipes/mfast/all/patches/00005-add-conan-libs-to-targets.patch.txt:9]: I hope we can solve it soon with the components feature ... You needed to create a big patch only to solve a problem which should be solved by Conan generators ...

File: recipes/mfast/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.8.11)
+project(PackageTest CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+FASTTYPEGEN_TARGET(MDRefreshSample MDRefreshSample.xml)
```
Comment (uilianries) [inline recipes/mfast/all/test_package/CMakeLists.txt:7]: I was thinking why mFastTools.cmake is so important ... question answered here.

File: recipes/mfast/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class mFASTTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    requires = "boost/1.73.0"
```
Comment (uilianries) [inline recipes/mfast/all/test_package/conanfile.py:8]:
```suggestion
```

Do not pass transitive requirements to test_package.

File: recipes/mfast/all/test_package/message_printer.h [hunk 7]
```diff
@@ -0,0 +1,139 @@
+#ifndef MESSAGE_PRINTER_H
```
Comment (uilianries) [inline recipes/mfast/all/test_package/message_printer.h:1]: Please, try to use something simpler, small. The test package is used for checking the package, not to exercise the project. Code examples are goods most of time, but this one is too big, includes boost as external dependency ...

File: recipes/mfast/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
+from conans import ConanFile, CMake, tools
+import os
+import shutil
+
+
+class mFASTConan(ConanFile):
+    name = "mfast"
+    license = "LGPL-3.0"
```
Comment (uilianries) [inline recipes/mfast/all/conanfile.py:8]:
```suggestion
    license = "BSD-3-Clause"
```

https://github.com/objectcomputing/mFAST/blob/master/licence.txt

Comment (tarc) [reply recipes/mfast/all/conanfile.py:8]: Ah! They changed the  license at some point. Version 1.2.1 I think is still LGPLv3:

https://github.com/objectcomputing/mFAST/blob/v1.2.1/licence.txt

Is it better to leave it as it is now or to update already the license to match the most updated one?

File: recipes/mfast/all/patches/00005-add-conan-libs-to-targets.patch.txt (hunk 4 above)
Comment (tarc) [reply recipes/mfast/all/patches/00005-add-conan-libs-to-targets.patch.txt:9]: You mean this feature:

https://docs.conan.io/en/latest/creating_packages/package_information.html#using-components

?

File: recipes/mfast/all/test_package/message_printer.h (hunk 7 above)
Comment (tarc) [reply recipes/mfast/all/test_package/message_printer.h:1]: Yeah, sorry for that. I forgot to trim this example down after I copied it from the examples. I'll be doing this tomorrow at lunch probably.

File: recipes/mfast/all/patches/00005-add-conan-libs-to-targets.patch.txt (hunk 4 above)
Comment (uilianries) [reply recipes/mfast/all/patches/00005-add-conan-libs-to-targets.patch.txt:9]: https://github.com/conan-io/conan-center-index/pull/2097

File: recipes/mfast/all/conanfile.py (hunk 8 above)
Comment (uilianries) [reply recipes/mfast/all/conanfile.py:8]: Indeed. Keep LGPL-3.0, for future versions we can use dynamic license.

File: recipes/mfast/all/test_package/message_printer.h (hunk 7 above)
Comment (tarc) [reply recipes/mfast/all/test_package/message_printer.h:1]: I think it's passable now, please confirm...

File: recipes/mfast/all/test_package/MDRefreshSample.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0"?>
```
Comment (uilianries) [inline recipes/mfast/all/test_package/MDRefreshSample.xml:1]: I think you can remove it now.

Comment (tarc) [reply recipes/mfast/all/test_package/MDRefreshSample.xml]: @uilianries , I updated the test package. Would you mind reviewing it another time?

I really simplified it now, but kept the xml (although reduced to a couple of tags) to verify that the CMake macro was successfully exported via the `cpp_info.build_modules` mechanism.

**Pull request merged**